mlir/test/python/dialects/arith_dialect.py

```diff
@@ -35,14 +36,57 @@ def testFastMathFlags():
             print(r)
 
 
-# CHECK-LABEL: TEST: testArithValueBuilder
+# CHECK-LABEL: TEST: testArithValue
 @run
-def testArithValueBuilder():
+def testArithValue():
+    def _binary_op(lhs, rhs, op: str) -> "ArithValue":
+        op = op.capitalize()
+        if arith._is_float_type(lhs.type):
```

Comment:
Should we also check the RHS? Otherwise we may allow arithmetic between floats and integers, which likely isn't something we want.